Document element type returned by `Array::zeros`

Doing something like `let xs = Array::zeros((100, 3));`, it's not clear from the API or [documentation for zeros](https://docs.rs/ndarray/latest/ndarray/struct.ArrayBase.html#method.zeros) what the return type is. Ie. are the elements integer, f64, f32 etc.? It would be good to document this, alongside the right way to construct arrays of zeros with different element datatypes.

The above does not compile exactly due to this:

```
error[E0283]: type annotations needed for `ArrayBase<OwnedRepr<A>, Dim<[usize; 2]>>`
   --> src/main.rs:2:9
    |
2   |     let xs = ndarray::Array::zeros((2, 3));
    |         ^^   --------------------- type must be known at this point
    |
    = note: cannot satisfy `_: Clone`
note: required by a bound in `ndarray::impl_constructors::<impl ArrayBase<S, D>>::zeros`
   --> /home/adam/.cargo/registry/src/github.com-1ecc6299db9ec823/ndarray-0.15.6/src/impl_constructors.rs:342:12
    |
342 |         A: Clone + Zero,
    |            ^^^^^ required by this bound in `ndarray::impl_constructors::<impl ArrayBase<S, D>>::zeros`
help: consider specifying the type argument in the function call
    |
2   |     let xs = ndarray::Array::zeros::<Sh>((2, 3));
    |                                   ++++++

For more information about this error, try `rustc --explain E0283`.
```

Meaning that one needs to constrain the element type somehow in any case, e.g.

```rust
let xs = ndarray::Array::<f64, _>::zeros((2, 3));
```

or

```rust
let xs: ndarray::Array<f64, _> = ndarray::Array::zeros((2, 3));
```

--- a/src/impl_constructors.rs
+++ b/src/impl_constructors.rs
@@ -340,15 +340,15 @@ where
     /// The element type is inferred; to control it, you can either specify
     /// type of the returned array or use turbofish syntax in the function call:
     /// ```
-    /// use ndarray::{Array1, Array2};
+    /// use ndarray::Array1;
     ///
     /// // Specify f32
     /// let arr_f32: Array1<f32> = Array1::zeros(3);
-    /// assert_eq!(arr_f32, [0_f32, 0, 0]);
+    /// assert_eq!(arr_f32, [0_f32, 0.0, 0.0]);
     ///
     /// // Specify i64
-    /// let arr_i64 = Array2::<i64>::zeros((2, 2));
-    /// assert_eq!(arr_i64, [[0_i64, 0], [0, 0]]);
+    /// let arr_i64 = Array1::<i64>::zeros(3);
+    /// assert_eq!(arr_i64, [0_i64, 0, 0]);
     /// ```
     ///
     /// **Panics** if the product of non-zero axis lengths overflows `isize`.

--- a/src/impl_constructors.rs
+++ b/src/impl_constructors.rs
@@ -337,6 +337,20 @@ where
 
     /// Create an array with zeros, shape `shape`.
     ///
+    /// The element type is inferred; to control it, you can either specify
+    /// type of the returned array or use turbofish syntax in the function call:
+    /// ```
+    /// use ndarray::{Array1, Array2};
+    ///
+    /// // Specify f32
+    /// let arr_f32: Array1<f32> = Array1::zeros(3);
+    /// assert_eq!(arr_f32, [0_f32, 0, 0]);
+    ///
+    /// // Specify i64
+    /// let arr_i64 = Array2::<i64>::zeros((2, 2));
+    /// assert_eq!(arr_i64, [[0_i64, 0], [0, 0]]);
+    /// ```
+    ///
     /// **Panics** if the product of non-zero axis lengths overflows `isize`.
     pub fn zeros<Sh>(shape: Sh) -> Self
     where

--- a/src/impl_constructors.rs
+++ b/src/impl_constructors.rs
@@ -340,15 +340,15 @@ where
     /// The element type is inferred; to control it, you can either specify
     /// type of the returned array or use turbofish syntax in the function call:
     /// ```
-    /// use ndarray::Array1;
+    /// use ndarray::{Array1, arr1};
     ///
     /// // Specify f32
     /// let arr_f32: Array1<f32> = Array1::zeros(3);
-    /// assert_eq!(arr_f32, [0_f32, 0.0, 0.0]);
+    /// assert_eq!(arr_f32, arr1(&[0_f32, 0.0, 0.0]));
     ///
     /// // Specify i64
     /// let arr_i64 = Array1::<i64>::zeros(3);
-    /// assert_eq!(arr_i64, [0_i64, 0, 0]);
+    /// assert_eq!(arr_i64, arr1(&[0_i64, 0, 0]));
     /// ```
     ///
     /// **Panics** if the product of non-zero axis lengths overflows `isize`.

